endpoints.php

```diff
@@ -373,24 +373,51 @@ function getTapestry($request)
 function addTapestry($request)
 {
     $tapestryData = json_decode($request->get_body());
-    if (!get_page_by_title($tapestryData->title, 'OBJECT', 'tapestry')) {
+    try {
+        if (get_page_by_title($tapestryData->title, 'OBJECT', 'tapestry')) {
+            throw new TapestryError('POST_EXISTS');
+        }
```

Comment:
I don't think this is required. Users should be allowed to add Tapestries that have the same title.